README.md

```diff
@@ -1,143 +1,105 @@
+1. Test Your Change
+1. Commit Your Change
+1. (Optional) Share How This Change Looks like to Others
+1. Publish Your Change to Your Github Account
+1. Create a Merge Request and Participate the Review
 
-### fork the source codes
+### Fork the Repository
 
-Firstly, let's get the source codes. We will "copy" this source codes to your github account. The term "fork" means to get a derived copy which contents are the same as the source.
+First, you need to get the source of the website by "forking" it. That is, to get a derived copy which share the history of the source repository.
 
-Click **fork** on the up-right of this [cesehub](https://github.com/cesehub/cesehub/) github repository page to fork the source codes to your github account. If you have not github account, please regist one for you first.
+Click **Fork** button on the top-right of this [cesehub](https://github.com/cesehub/cesehub/) github page to fork it to your github account. (You need a github account ready.)
 
-### clone the source codes to your local machine
+### Clone the Forked Repository to Your Local Machine
 
-Clone the source codes to your local machine, so you could edit the codes on your laptop or desktop.
-In this session, we will get two working copies. One is a folder with the name **cesehub** and the other is **cesehub-dev**.
+You have to clone the forked repository to your local machine before you can edit the source code of the website. In the example of this section, we will get two working copies. One is a folder with the name `cesehub` and the other is `cesehub-dev`.
 
-On your github page holding your cesehub fork which usually looks like https://github.com/<your account name>/cesehub/, there is a button **Clone or download**. Click it and copy the url of **Clone with HTTPS** to get your-repo-url which will be used later.
+The URL of your cesehub repository on github should look like `https://github.com/<your account name>/cesehub/`. There is a button **Clone or download**. Click it and copy the URL of **Clone with HTTPS** to get `<your-repo-url>` which will be used later.
 
-Please open your termial application of your machine, and type in the following command.
+Please open your termial application of your machine, and type in the following command:
 
 ```
 git clone <your-repo-url>
 ```
-you-repo-url is the url you just got on your github page. Your system needs to install git first to use this application.
-Now you should have a folder named **cesehub** on your local machine.
-This repository will be used to host your fork github page. You could share the url it hosts to the others for their reviews. We will talk about the url later.
-We called this is a staging site.
-
-Finally, clone the source code directly from the cesehub upstream
-```
-git clone https://github.com/cesehub/cesehub/ cesehub-dev
-```
-
-You should have two folders on your local machine now. One is cesehub and another is cesehub-dev.
 
-##### remove CNAME
```

Comment:
## Why don't we discuss the topic right here?

We could still use this thread to discuss the CNAME topic. The following reasons let me decide to merge this pull request first:

* I think the admin/contributor point of view is convincing enough to me to accept a PR. Because the version without removing CNAME is acceptable to me, it is an optional session.
* Merging the PR could let me have no need to review all the other change, which I think they are good enough and distracts me when I tried to focus on CNAME topic. I prefer to make some changes based on most of this PR has done. For example, [this change](https://github.com/cesehub/cesehub/pull/19/files).

## Are you proposing to postpone the discussion?
No. I want to make the further discussion be more readable to me.

## Continue the CNAME topic discssion
CNAME is removed in the gh-pages branch of the contributors's own repository. This branch is not used to make a pull request. The branch used to make a pull request from a contributor is `dev`. Please do not worry about that removing CNAME in the contributors's repository will cause difficulty to merge into the trunk. 

I prefer to provide a flow for a contributor to host his/her own customized CESE Hub site by github, for example, like this http://tai271828.github.io/cesehub/, so he/she could easily share how this site looks like if he/she wants. When a content contributor tries to do this, he/she needs to push their code to the gh-pages branch of their own forked repository. This will cause two issues:
* They will keep getting warning spam emails with the subject "Page build warning" from github. That is very annoyed.
* _config.yml needs to be fixed at the same time to make the github-hosted customized CESE Hub could access associated  resources like images. From the admin/contributor point of view, this file should be edited by the admin as well.

From the contributor's repository point of view, the admin of his/her gh-branches (and their customized CESE Hub website) is himself/herself. They could definitely touch CNAME and _config.yml. The only thing they need to note is not to mix this change with the change which they used for a pull request.

[This PR](https://github.com/cesehub/cesehub/pull/20/files) shows how the forementioned change looks like. Please note, this is an optional information. I prefer to provide this information, but it is not necessary in the mean time.